### ISSUE TITLE
Build stage configuration flag "enabled" for provider codebuild not working properly 2

### ISSUE THREAD

**Issue by lordoflairs** [awslabs/aws-deployment-framework#236]
Using pipeline with just source codecommit and then deploy with cloudformation by setting enabled to false for the build step will generate output from source not match input to cloudformation. (InputArtifacts to cloudformation does not match OutputArtifacts from previous step.)

The problem is probably in adf_codepipeline.py but I am not bright enough to figure it out. (Not a Python coder...)

E.g. changeset will be:

```
        "Stages": [
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "AWS",
                  "Provider": "CodeCommit",
                  "Version": "1"
                },
                "Configuration": {
                  "BranchName": "master",
                  "RepositoryName": "myRepo",
                  "PollForSourceChanges": false
                },
                "Name": "source",
                "OutputArtifacts": [
                  {
                    "Name": "output-source"
                  }
                ],
                "Region": "eu-west-1",
                "RoleArn": "arn:aws:iam::XXXXXX:role/adf-codecommit-role",
                "RunOrder": 1
              }
            ],
            "Name": "Source-XXXXXX"
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "CloudFormation",
                  "Version": "1"
                },
                "Configuration": {
                  "ActionMode": "CHANGE_SET_REPLACE",
                  "StackName": "adf-stack-name",
                  "ChangeSetName": "adf-stack-name",
                  "TemplateConfiguration": "stack-name-build::params/CloudCI_eu-west-1.json",
                  "Capabilities": "CAPABILITY_NAMED_IAM,CAPABILITY_AUTO_EXPAND",
                  "RoleArn": "arn:aws:iam::XXXXXX:role/adf-cloudformation-deployment-role",
                  "TemplatePath": "stack-name-build::template.yml"
                },
                "InputArtifacts": [
                  {
                    "Name": "stack-name-build"
                  }
                ],
                "Name": "CloudCI-eu-west-1-create",
                "Region": "eu-west-1",
                "RoleArn": "arn:aws:iam::XXXXXX:role/adf-cloudformation-role",
                "RunOrder": 1
              },
```

**Comment by lordoflairs** [comment]
Maybe there are more issues to this...
"TemplateConfiguration": "stack-name-build::params/CloudCI_eu-west-1.json"
I guess this means that params are supposed to be generated in build step?

**Comment by sbkok** [comment]
Hi @lordoflairs, thanks reaching out!
@bundyfx tried this in his account and it fails as you described.

For anyone else trying to replicate this, the deployment map specification would look like this:

```yaml
  - name: sample-pipeline-wo-codebuild
    default_providers:
      source:
        provider: codecommit
        properties:
          account_id: 111111111111
      build:
        properties:
          enabled: False
    targets:
      - /some/target
```

In ADFs design, it uses the `params` that you referred to in the last comments to apply tags and generate parameters on an account-by-account basis. When you would add a new account to the OU it targets, it would automatically get the right parameters this way. This step always executes in CodeBuild before it is deployed by CloudFormation.

What are you trying to accomplish here such that it wouldn't need this step?

**Comment by lordoflairs** [comment]
Hello and thank you for helping out.
We had a simple pipeline that did not need any parameter input.
To me this does not seem like a to uncommon scenario?

For us it was a deploy of a lambda function in the deploy account to relay sns topic information to a teams channel but any infrastructure template.yml that does not require specific parameters would have the same setup.

**Comment by lordoflairs** [comment]
For now the workaround is to have a buildstep that have to copy from $S3_BUCKET_NAME and call adf-build/generate_params.py just to genrate an empty parameter file.

**Comment by bundyfx** [comment]
Thanks @lordoflairs - this will be resolved in 3.0.5 coming shortly.
